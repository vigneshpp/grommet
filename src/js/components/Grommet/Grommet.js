import React, { Component } from 'react';
import { createGlobalStyle } from 'styled-components';

import {
  ResponsiveContext,
  ThemeContext,
  ContainerTargetContext,
} from '../../contexts';

import {
  backgroundIsDark,
  deepMerge,
  getBreakpoint,
  getDeviceBreakpoint,
  normalizeColor,
} from '../../utils';
import { base as baseTheme } from '../../themes';
import { StyledGrommet } from './StyledGrommet';

const FullGlobalStyle = createGlobalStyle`
  body { margin: 0; }
`;

class Grommet extends Component {
  static displayName = 'Grommet';

  static getDerivedStateFromProps(nextProps, prevState) {
    const {
      background: backgroundProp,
      dir,
      theme = {},
      themeMode,
    } = nextProps;
    const { theme: stateTheme, themeProp, themeModeProp } = prevState;

    const nextTheme = deepMerge(baseTheme, theme);
    if (!stateTheme || theme !== themeProp || themeMode !== themeModeProp) {
      const {
        colors: { background: themeBackground },
      } = nextTheme.global;

      // get initial value for dark so we can normalize background color
      nextTheme.dark = (themeMode || theme.defaultMode) === 'dark';
      const color = normalizeColor(
        backgroundProp || themeBackground,
        nextTheme,
      );

      // After normalizing, we set nextTheme.dark once more.
      // It is necessary that we set it twice. We have to handle two cases:
      // 1. Caller passes in a color object or a color name that resolves an
      //    object. In this case, we want to set dark as line 38 shows. The
      //    second set, in line 46, is a no-op.
      // 2. Caller passes a specific color value or a color name that resolves
      //    to a specific color value. In this case, we want dark to be set
      //    based on that color, which line 46 will do.
      // The double set of nextTheme.dark allows us to handle both cases here
      // without having to duplicate color object + name + dark mode detection
      // code here that is already in normalizeColor and backgroundIsDark.
      nextTheme.dark = backgroundIsDark(color, nextTheme);
      nextTheme.baseBackground = backgroundProp || themeBackground;

      if (dir) {
        nextTheme.dir = dir;
      }
      return {
        theme: nextTheme,
        themeProp: theme,
        themeModeProp: themeMode,
      };
    }

    return null;
  }

  state = {};

  componentDidMount() {
    window.addEventListener('resize', this.onResize);
    this.onResize();
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.onResize);
  }

  onResize = () => {
    const { theme, responsive } = this.state;

    let breakpoint;
    // responsive would be undefined in the case of SSR or initial page load
    if (!responsive) {
      // In the case of SSR we'll need to use the user agent breakpoint
      breakpoint = this.deviceResponsive();
    }
    // Initial page load where both responsive and breakpoint are undefined
    if (!breakpoint) breakpoint = getBreakpoint(window.innerWidth, theme);

    if (breakpoint !== responsive) {
      this.setState({ responsive: breakpoint });
    }
  };

  deviceResponsive() {
    const { userAgent } = this.props;
    const { theme } = this.state;

    /*
     * Regexes provided for mobile and tablet detection are meant to replace
     * a full-featured specific library due to contributing a considerable size
     * into the bundle.
     *
     * User agents found https://deviceatlas.com/blog/list-of-user-agent-strings
     */
    if (userAgent) {
      if (
        /(tablet|ipad|playbook|silk)|(android(?!.*mobile))/i.test(userAgent)
      ) {
        return getDeviceBreakpoint('tablet', theme);
      }
      if (/Mobile|iPhone|Android/.test(userAgent)) {
        return getDeviceBreakpoint('phone', theme);
      }
      return getDeviceBreakpoint('computer', theme);
    }
    return undefined;
  }

  render() {
    const {
      children,
      full,
      containerTarget = typeof document === 'object'
        ? document.body
        : undefined,
      ...rest
    } = this.props;
    delete rest.theme;
    const { theme, responsive: stateResponsive } = this.state;

    // Value from state should be correct once we resize
    // On first render we try to guess otherwise set the default as a tablet
    const responsive =
      stateResponsive ||
      this.deviceResponsive() ||
      theme.global.deviceBreakpoints.tablet;

    return (
      <ThemeContext.Provider value={theme}>
        <ResponsiveContext.Provider value={responsive}>
          <ContainerTargetContext.Provider value={containerTarget}>
            <StyledGrommet full={full} {...rest}>
              {children}
            </StyledGrommet>
            {full && <FullGlobalStyle />}
          </ContainerTargetContext.Provider>
        </ResponsiveContext.Provider>
      </ThemeContext.Provider>
    );
  }
}

let GrommetDoc;
if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  GrommetDoc = require('./doc').doc(Grommet);
}
const GrommetWrapper = GrommetDoc || Grommet;

export { GrommetWrapper as Grommet };
