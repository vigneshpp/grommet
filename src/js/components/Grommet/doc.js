import { describe, PropTypes } from 'react-desc';

import { backgroundDoc } from '../../utils/prop-types';
import { getAvailableAtBadge } from '../../utils/mixins';

export const doc = Grommet => {
  const DocumentedGrommet = describe(Grommet)
    .availableAt(getAvailableAtBadge('Grommet', 'Utilities'))
    .description('The top level Grommet container.')
    .usage(
      `import { Grommet } from 'grommet';
<Grommet>...</Grommet>`,
    )
    .intrinsicElement('div');

  DocumentedGrommet.propTypes = {
    background: backgroundDoc,
    dir: PropTypes.oneOf(['rtl']).description(
      'Layout direction for right to left contexts',
    ),
    full: PropTypes.bool
      .description('Whether to take the whole viewport.')
      .defaultValue(false),
    plain: PropTypes.bool
      .description(
        `Whether or not Grommet should apply a global font-family, font-size,
        and line-height.`,
      )
      .defaultValue(false),
    cssVars: PropTypes.bool
      .description('Whether to expose the css variables.')
      .defaultValue(false),
    theme: PropTypes.object.description(
      'Custom styles for Grommet app component.',
    ),
    themeMode: PropTypes.oneOf(['dark', 'light']).description(
      `Dark vs. light theme variation. Default is unspecified and left to
      theme.`,
    ),
    userAgent: PropTypes.string.description(
      `User agent used to detect the device width for setting the initial
      breakpoint.`,
    ),
    containerTarget: PropTypes.object.description(
      `The node where Drop and Layer containers are inserted. Defaults to
      document.body which is almost always the right choice. This is used
      for less common cases like rendering within an internal node (e.g.
      shadow root).`,
    ),
  };

  return DocumentedGrommet;
};

export const themeDoc = {
  'grommet.extend': {
    description: 'Any additional style for Grommet.',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
  'global.font.face': {
    description: 'Custom font face declaration',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
};
