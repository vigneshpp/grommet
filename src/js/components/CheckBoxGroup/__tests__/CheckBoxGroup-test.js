import React from 'react';
import renderer from 'react-test-renderer';

import 'jest-axe/extend-expect';
import 'regenerator-runtime/runtime';

import { cleanup, render, fireEvent } from '@testing-library/react';
import { axe } from 'jest-axe';
import { Grommet } from '../../Grommet';
import { CheckBoxGroup } from '..';

describe('CheckBoxGroup', () => {
  afterEach(cleanup);

  test('should have no accessibility violations', async () => {
    const { container } = render(
      <Grommet>
        <CheckBoxGroup options={['First', 'Second']} />
      </Grommet>,
    );

    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  test('options renders', () => {
    const component = renderer.create(
      <Grommet>
        <CheckBoxGroup options={['First', 'Second']} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('value renders', () => {
    const component = renderer.create(
      <Grommet>
        <CheckBoxGroup value={['First']} options={['First', 'Second']} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('initial value renders', () => {
    const component = renderer.create(
      <Grommet>
        <CheckBoxGroup
          value={['Wuhan', 'Jerusalem']}
          options={[
            { label: 'Maui', value: 'Maui' },
            { label: 'Jerusalem', value: 'Jerusalem' },
            { label: 'Wuhan', value: 'Wuhan' },
          ]}
        />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('disabled renders', () => {
    const component = renderer.create(
      <Grommet>
        <CheckBoxGroup disabled options={['First', 'Second']} />
        <CheckBoxGroup options={[{ label: 'First', disabled: true }]} />
        <CheckBoxGroup options={[{ label: 'First', disabled: true }]} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('onChange', () => {
    const onChange = jest.fn();
    const { container, getByText } = render(
      <Grommet>
        <CheckBoxGroup
          options={[
            { label: 'first-label', value: 'First' },
            { label: 'second-label', value: 'Second' },
          ]}
          onChange={onChange}
        />
      </Grommet>,
    );
    fireEvent.click(getByText('first-label'));
    expect(onChange).toBeCalledTimes(1);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('onClick for check and uncheck a CheckBox', () => {
    const onClick = jest.fn();
    const { container, getByText } = render(
      <Grommet>
        <CheckBoxGroup
          options={[
            { label: 'first-label', value: 'First' },
            { label: 'second-label', value: 'Second' },
          ]}
          onClick={onClick}
        />
      </Grommet>,
    );
    fireEvent.click(getByText('first-label'));
    expect(onClick).toBeCalledTimes(1);
    expect(container.firstChild).toMatchSnapshot();
    fireEvent.click(getByText('first-label'));
    expect(onClick).toBeCalledTimes(2);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('labelKey', () => {
    const { container } = render(
      <Grommet>
        <CheckBoxGroup
          labelKey="labelKeyTest"
          options={[
            { labelKeyTest: 'first-label', value: 'First' },
            { labelKeyTest: 'second-label', value: 'Second' },
          ]}
        />
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('valueKey', () => {
    const { container } = render(
      <Grommet>
        <CheckBoxGroup
          valueKey="valueKeyTest"
          options={[
            { label: 'first-label', valueKeyTest: 'First' },
            { label: 'second-label', valueKeyTest: 'Second' },
          ]}
        />
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('checked warning', () => {
    console.warn = jest.fn();
    const warnSpy = jest.spyOn(console, 'warn');

    render(
      <Grommet>
        <CheckBoxGroup
          options={[
            { label: 'first-label', value: 'First', checked: true },
            { label: 'second-label', value: 'Second' },
          ]}
        />
      </Grommet>,
    );

    expect(warnSpy).toBeCalled();
    warnSpy.mockReset();
    warnSpy.mockRestore();
    console.warn.mockReset();
  });
});
