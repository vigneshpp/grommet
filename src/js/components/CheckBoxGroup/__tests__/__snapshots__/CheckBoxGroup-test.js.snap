// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxGroup disabled renders 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 lcXYAf"
>
  <div
    className="StyledBox-sc-13pk1d4-0 bFwQzJ"
  >
    <label
      checked={false}
      className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 dEVPVv"
      disabled={true}
      onClick={[Function]}
    >
      <div
        checked={false}
        className="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
        disabled={true}
      >
        <input
          checked={false}
          className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 gDJpaa"
          disabled={true}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <div
          checked={false}
          className="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
          disabled={true}
        />
      </div>
      <span>
        First
      </span>
    </label>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
    />
    <label
      checked={false}
      className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 dEVPVv"
      disabled={true}
      onClick={[Function]}
    >
      <div
        checked={false}
        className="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
        disabled={true}
      >
        <input
          checked={false}
          className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 gDJpaa"
          disabled={true}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <div
          checked={false}
          className="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
          disabled={true}
        />
      </div>
      <span>
        Second
      </span>
    </label>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 bFwQzJ"
  >
    <label
      checked={false}
      className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 dEVPVv"
      disabled={true}
      onClick={[Function]}
    >
      <div
        checked={false}
        className="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
        disabled={true}
      >
        <input
          checked={false}
          className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 gDJpaa"
          disabled={true}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <div
          checked={false}
          className="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
          disabled={true}
        />
      </div>
      <span>
        First
      </span>
    </label>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 bFwQzJ"
  >
    <label
      checked={false}
      className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 dEVPVv"
      disabled={true}
      onClick={[Function]}
    >
      <div
        checked={false}
        className="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
        disabled={true}
      >
        <input
          checked={false}
          className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 gDJpaa"
          disabled={true}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <div
          checked={false}
          className="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
          disabled={true}
        />
      </div>
      <span>
        First
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup initial value renders 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 lcXYAf"
>
  <div
    className="StyledBox-sc-13pk1d4-0 bFwQzJ"
  >
    <label
      checked={false}
      className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
      onClick={[Function]}
    >
      <div
        checked={false}
        className="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          checked={false}
          className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <div
          checked={false}
          className="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        />
      </div>
      <span>
        Maui
      </span>
    </label>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
    />
    <label
      checked={true}
      className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
      onClick={[Function]}
    >
      <div
        checked={true}
        className="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          checked={true}
          className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <div
          checked={true}
          className="StyledBox-sc-13pk1d4-0 KsQmP StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        >
          <svg
            checked={true}
            className="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 gHRhNW"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,11.3 L10.3,16 L18,6.2"
              fill="none"
            />
          </svg>
        </div>
      </div>
      <span>
        Jerusalem
      </span>
    </label>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
    />
    <label
      checked={true}
      className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
      onClick={[Function]}
    >
      <div
        checked={true}
        className="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          checked={true}
          className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <div
          checked={true}
          className="StyledBox-sc-13pk1d4-0 KsQmP StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        >
          <svg
            checked={true}
            className="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 gHRhNW"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,11.3 L10.3,16 L18,6.2"
              fill="none"
            />
          </svg>
        </div>
      </div>
      <span>
        Wuhan
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup labelKey 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 lcXYAf"
>
  <div
    class="StyledBox-sc-13pk1d4-0 bFwQzJ"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        />
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup onChange 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 lcXYAf"
>
  <div
    class="StyledBox-sc-13pk1d4-0 bFwQzJ"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 KsQmP StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        >
          <svg
            class="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 gHRhNW"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,11.3 L10.3,16 L18,6.2"
              fill="none"
            />
          </svg>
        </div>
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup onClick for check and uncheck a CheckBox 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 lcXYAf"
>
  <div
    class="StyledBox-sc-13pk1d4-0 iDvhFF"
    tabindex="0"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 KsQmP StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        >
          <svg
            class="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 gHRhNW"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,11.3 L10.3,16 L18,6.2"
              fill="none"
            />
          </svg>
        </div>
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup onClick for check and uncheck a CheckBox 2`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 lcXYAf"
>
  <div
    class="StyledBox-sc-13pk1d4-0 iDvhFF"
    tabindex="0"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        />
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup options renders 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 lcXYAf"
>
  <div
    className="StyledBox-sc-13pk1d4-0 bFwQzJ"
  >
    <label
      checked={false}
      className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
      onClick={[Function]}
    >
      <div
        checked={false}
        className="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          checked={false}
          className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <div
          checked={false}
          className="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        />
      </div>
      <span>
        First
      </span>
    </label>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
    />
    <label
      checked={false}
      className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
      onClick={[Function]}
    >
      <div
        checked={false}
        className="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          checked={false}
          className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <div
          checked={false}
          className="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        />
      </div>
      <span>
        Second
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup value renders 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 lcXYAf"
>
  <div
    className="StyledBox-sc-13pk1d4-0 bFwQzJ"
  >
    <label
      checked={true}
      className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
      onClick={[Function]}
    >
      <div
        checked={true}
        className="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          checked={true}
          className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <div
          checked={true}
          className="StyledBox-sc-13pk1d4-0 KsQmP StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        >
          <svg
            checked={true}
            className="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 gHRhNW"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,11.3 L10.3,16 L18,6.2"
              fill="none"
            />
          </svg>
        </div>
      </div>
      <span>
        First
      </span>
    </label>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
    />
    <label
      checked={false}
      className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
      onClick={[Function]}
    >
      <div
        checked={false}
        className="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          checked={false}
          className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <div
          checked={false}
          className="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        />
      </div>
      <span>
        Second
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup valueKey 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 lcXYAf"
>
  <div
    class="StyledBox-sc-13pk1d4-0 bFwQzJ"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        />
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fuEVwc StyledCheckBox-sc-1dbk5ju-6 CBoMo"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 gDUCoH StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;
